src/conn/conn_tiered.c

```diff
@@ -0,0 +1,328 @@
+/*-
+ * Copyright (c) 2014-2020 MongoDB, Inc.
+ * Copyright (c) 2008-2014 WiredTiger, Inc.
+ *	All rights reserved.
+ *
+ * See the file LICENSE for redistribution information.
+ */
+
+#include "wt_internal.h"
+
+#ifdef __GNUC__
+#if __GNUC__ > 4 || (__GNUC__ == 4 && __GNUC_MINOR__ > 1)
+/*
+ * !!!
+ * GCC with -Wformat-nonliteral complains about calls to strftime in this file.
+ * There's nothing wrong, this makes the warning go away.
+ */
+#pragma GCC diagnostic ignored "-Wformat-nonliteral"
+#endif
+#endif
+
+/*
+ * __tier_storage_once --
+ *     Perform one iteration of tiered storage maintenance.
+ */
+static void
+__tier_storage_once(WT_SESSION_IMPL *session, bool force)
+{
+    WT_UNUSED(session);
+    WT_UNUSED(force);
+    /*
+     * - See if there is any "merging" work to do to prepare and create an object that is
+     *   suitable for placing onto tiered storage.
+     * - Do the work to create said objects.
+     * - Move the objects.
+     */
+    return;
+}
+
+/*
+ * __tier_storage_remove_tree --
+ *     Perform one iteration of tiered storage local tier removal.
+ */
+static int
+__tier_storage_remove_tree(WT_SESSION_IMPL *session, const char *uri, bool force)
+{
+    WT_CONFIG_ITEM cval;
+    WT_DECL_RET;
+    size_t len;
+    uint64_t now;
+    char *config, *newfile;
+    const char *cfg[2], *filename;
+
+    if (uri == NULL)
+        return (0);
+    __wt_verbose(session, WT_VERB_TIERED, "Removing tree %s", uri);
+    WT_ASSERT(session, WT_PREFIX_MATCH(uri, "tiered:"));
+    filename = uri;
+    WT_PREFIX_SKIP_REQUIRED(session, filename, "tiered:");
+    len = strlen("file:") + strlen(filename) + 1;
+    WT_ERR(__wt_calloc_def(session, len, &newfile));
+    WT_ERR(__wt_snprintf(newfile, len, "file:%s", filename));
+
+    /*
+     * If the file version of the tiered object does not exist there is nothing to do.
+     */
+    WT_ERR(__wt_metadata_search(session, newfile, &config));
```

Comment:
Is the comment referring to the metadata for the file not existing?  Or the file itself?  If the metadata doesn't exist, this function will `WT_NOTFOUND`.  If the comment refers to the file, please move it to after the metadata search, otherwise it seems misleading.

Reply:
We come in with a URI that is "tiered:name" and build "file:name" that is "the file version of the tiered object" that I'm referring to. The code is doing a metadata search of the file:URI at that line. So if that file:URI does not exist, then we have no more work to do. I will slightly change the comment to make that clearer.

Reply:
So if the metadata `"file:name"` doesn't exist, doesn't it return `WT_NOTFOUND`? Won't that be considered an error by the caller?

Reply:
Ah, thank you for pushing on that more. I will not have it return the error indicator.